justbar/src/main/java/com/irozon/justbar/BarItem.java

```diff
@@ -44,274 +43,286 @@ public boolean isSelected() {
-        RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
-        layoutParams.addRule(RelativeLayout.CENTER_IN_PARENT); // A position in layout.
-
-        imageView = new ImageView(context);
-        imageView.setLayoutParams(layoutParams);
-
-        if (icon != null)
-            imageView.setImageDrawable(icon);
-
-        addView(imageView);
-
-        if (selected)
+        DependentLayout.LayoutConfig layoutParams = new DependentLayout
+                .LayoutConfig(ComponentContainer.LayoutConfig.MATCH_CONTENT,
+                ComponentContainer.LayoutConfig.MATCH_CONTENT);
+        layoutParams.addRule(LayoutConfig.CENTER_IN_PARENT); // A position in layout.
+        imageView = new Image(context);
+        imageView.setLayoutConfig(layoutParams);
+        if (icon != null) {
+            imageView.setImageElement(icon);
+        }
+        addComponent(imageView);
+        if (selected) {
             makeSelected();
-
+        }
         setInitialColors();
     }
 
     /**
-     * Set initial color of the BarItem according to the attributes
+     * Set initial color of the BarItem according to the attributes.
      */
     private void setInitialColors() {
         if (selected) {
-            getBackground().setColorFilter(selectedColor, PorterDuff.Mode.SRC_IN);
-            imageView.setColorFilter(selectedIconColor);
+            getBackgroundElement().setColorMatrix(createColorMatrix(selectedColor));
+            getBackgroundElement().setStateColorMode(BlendMode.SRC_IN);
+            imageView.getImageElement().setColorMatrix(createColorMatrix(selectedIconColor));
         } else {
-            getBackground().setColorFilter(unSelectedColor, PorterDuff.Mode.SRC_IN);
-            imageView.setColorFilter(unSelectedIconColor);
+            getBackgroundElement().setColorMatrix(createColorMatrix(unSelectedColor));
+            getBackgroundElement().setStateColorMode(BlendMode.SRC_IN);
+            imageView.getImageElement().setColorMatrix(createColorMatrix(unSelectedIconColor));
         }
-
     }
 
     /**
-     * Get initial state of the BarItem (selected/unselected)
+     * Get initial state of the BarItem (selected/unselected).
      *
      * @param attrs AttributeSet
      * @return selected or unselected
      */
-    private boolean getSelectedStatus(AttributeSet attrs) {
-        TypedArray ta = context.obtainStyledAttributes(attrs, R.styleable.BarItem, 0, 0);
+    private boolean getSelectedStatus(AttrSet attrs) {
+        boolean selectedAttr = attrs.getAttr("selected").isPresent();
         try {
-            return ta.getBoolean(R.styleable.BarItem_selected, false);
+            return selectedAttr ? attrs.getAttr("selected").get().getBoolValue() : selectedAttr;
         } catch (Exception e) {
             return false;
         } finally {
-            ta.recycle();
+            //do nothing
```

Comment:
finally block can be removed as there is no implementation inside the block.

Reply:
Done